### DISCUSSION TITLE
How to collect all SQL (MySQL) of Java applications

### DISCUSSION THREAD

**Discussion by liuxinwang** [apache/skywalking#12655] [Q&A] [answered]
How to collect all SQL (MySQL) of Java applications to facilitate DBA audit.

**Comment by wu-sheng** [comment, marked as answer]
We have all the SQLs in the trace if you have adoptive SQL/DB plugins. But we don't have an exporter for them. I think you need to extend the agent analyzer and try to build that exporter.

**Comment by liuxinwang** [comment]
I didn't find SQL/DB plugins at https://skywalking.apache.org/docs/. Can you give me a document address?

**Reply by wu-sheng** [reply]
You get me wrong. I mean SQLs are collected by agent through tracing(this is agent SQL plugin). We don't have extra plugin to export them.

Audit is not feature scope of SkyWalking.

**Comment by liuxinwang** [comment]
Hello, I collect SQL through the beforeMethod method of the PreparedStatementExecuteMethodsInterceptor class in the agent plug-in mysql-common, and I can collect parameters normally, but the argumentsTypes in the input parameter cannot be obtained，The returned value is an empty array：before argumentsTypes:[]

**Reply by wu-sheng** [reply]
I don't know what do you mean. As I said, your use case is not our feature and not in our landscape.
So, you are on your own. The types are determined through defining instrumentation intercepting point. They are not for you to rebuilding the request.
